web/modules/custom/dpl_event/dpl_event.module

```diff
@@ -117,3 +118,28 @@ function dpl_event_set_occurred_callback(JobSchedule $job): void {
     $event->save();
   }
 }
+
+/**
+ * Implements hook_form_node_event_form_alter().
+ */
+function dpl_event_form_node_event_form_alter(array &$form, FormStateInterface $form_state, string $form_id): void {
+  // Use the generic "Access" value as the default for the first ticket
+  // category when creating events.
+  $form["field_ticket_categories"]["widget"][0]["subform"]["field_ticket_category_name"]["widget"][0]["value"]["#default_value"] = t('Access', [], ['context' => 'dpl_event']);
+}
+
+/**
+ * Implements hook_preprocess_field__field_ticket_category_price().
+ *
+ * Adds an array of formatted prices.
+ */
+function dpl_event_preprocess_field__field_ticket_category_price(array &$variables) : void {
+  /** @var \Drupal\dpl_event\PriceFormatter $price_formatter */
+  $price_formatter = \Drupal::service('dpl_event.price_formatter');
+  /** @var \Drupal\Core\Field\FieldItemListInterface $items */
```

Comment:
I do not think that is a good idea.

If we add an`instanceof`check this means that if a developer updates services definitions, then the code will still work technically but the correct formatting will not be used. It is kind of a silent failure. Since a change in services is initiated by a developer then it should also be easy for developers to become aware of any problems a change might cause. This will indeed be the case if rendering the event page fails. This is also be easy for any automated test to catch.

For user input it is another matter :smile:.